src/layout/strategy/collision.js

```diff
@@ -0,0 +1,88 @@
+export default function() {
+
+    var data = [];
+    var collision = function(inputData) {
```

Comment:
I don't think this really works as a component. The typical pattern is a [fluent builder pattern](http://blog.crisp.se/2013/10/09/perlundholm/another-builder-pattern-for-java) which constructs a function which is then either invoked with a selection or some data.

In this case you are returning something more complex that has various methods on it. 

I think this would be better as a simple set of utility functions - there is very little shared between utility method.